chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb

```diff
@@ -56,20 +56,26 @@
 
   include_recipe "crowbar::stop-services-before-upgrade"
 
+  dump_location = node[:crowbar][:upgrade][:db_dump_location]
+  save_node = false
+
+  unless node[:crowbar][:upgrade][:db_dump_location]
```

Comment:
Oh, hrm, your check won't work because the attribute is existing through the role. You probably need to check if it's set on the node directly. Or just rename the attribute?